test/iterators.jl

```diff
@@ -149,6 +149,30 @@ for xs in Any["abc", [1, 2, 3]]
     @test take(drop(take(xs, 3), 1), 1) === take(drop(xs, 1), 1)
 end
 
+# takewhile
+# --------
+@testset "takewhile" begin
+    @test takewhile(x -> x<4,1:10) |> collect == [1,2,3]
```

Comment:
```suggestion
    @test collect(takewhile(x -> x<4,1:10)) == [1,2,3]
```
And the rest of these in this file. Won't comment for each.

Reply:
The others haven't been addressed

Reply:
received. i have some work ahead for now, i will fix it tomorrow.